### PR TITLE
2597 remove proof

### LINKED ISSUE #2597: Remove `getOriginationProof` RPC method in Taquito
Protocol Oxford removes the RPC for computing smart rollups' origination proof.

- Remove the RPC method in `@taquito/rpc`, including rpc-cache, interface, etc
- Remove tests related to the origination proof RPC endpoint
- Remove it from smart rollup origination parameters in the Contract API, PrepareProvider, etc

relevant documents:
- https://tezos.gitlab.io/protocols/018_oxford.html#smart-rollups
- https://gitlab.com/tezos/tezos/-/merge_requests/8817

### PR DESCRIPTION
**Pull request by dsawali**
closes #2597 #2648 #2647 

- removed originationProof and its references in Taquito and in tests
- updated estimation values

Note that the failures in Oxfordnet tesnet integration tests are caused by the `whitelist` property addition in `smart_rollup_originate`. Ticket for reference: https://github.com/ecadlabs/taquito/issues/2776

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have run the linter against the changes
- [ ] You have added unit tests (if relevant/appropriate)
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [ ] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [ ] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: docs/smart_rollups.md [hunk 1]
```diff
@@ -29,7 +29,6 @@ await op.confirmation();
 - `pvmKind` at this time the only PVM supported is `wasm_2_0_0`
 - `kernel` is passed as a hexadecimal string examples can be found at [this tezos docs endpoint](https://tezos.gitlab.io/mumbai/smart_rollups.html?highlight=smart%20rollup%20originate#id1)
```
Comment (hui-an-yang) [inline docs/smart_rollups.md:30]: link is 404 might want to update it to `https://tezos.gitlab.io/active/smart_rollups.html`

File: integration-tests/contract-estimation-tests.spec.ts
Comment (hui-an-yang) [inline integration-tests/contract-estimation-tests.spec.ts:1]: thanks for taking care of this in this pr

File: integration-tests/data/metadataViews.ts
Comment (hui-an-yang) [inline integration-tests/data/metadataViews.ts:1]: thanks for taking care of this too

**Review by hui-an-yang — APPROVED** [review]
lgtm, looks like ticket#2647 #2648 can be closed by this pr too

File: docs/smart_rollups.md (hunk 1 above)
Comment (dsawali) [reply docs/smart_rollups.md:30]: good catch, fixing

**Pull request merged**